### PR TITLE
Remove Python 2.7 and 3.5 support, add Python 3.9 support

### PR DESCRIPTION
**Pull request by Gallaecio**
Once merged, we can also merge https://github.com/scrapy/w3lib/pull/167

To do:
- [x] [Comment for skipping W503](https://github.com/scrapy/w3lib/pull/168#discussion_r599352409)
- [x] [Review usage of `tests/py3-ignores.txt`](https://github.com/scrapy/w3lib/pull/168#pullrequestreview-618306637)
- [x] [Stop using the `u` prefix in literal strings](https://github.com/scrapy/w3lib/pull/168#discussion_r599353210)
- [x] [Remove encoding declarations](https://github.com/scrapy/w3lib/pull/168#discussion_r599354780)

### CODE REVIEW

**Review by elacuesta — APPROVED** [review]
Nice :muscle:

File: pytest.ini [hunk 1]
```diff
@@ -1,6 +1,8 @@
 [pytest]
 doctest_optionflags = ALLOW_UNICODE ALLOW_BYTES
 flake8-ignore =
+    W503
```
Comment (noviluni) [inline pytest.ini:4]: it probably deserves a comment

File: tests/test_html.py [hunk 2]
```diff
@@ -1,18 +1,25 @@
...
     def test_returns_unicode(self):
         # make sure it always return uncode
-        assert isinstance(replace_entities(b'no entities'), six.text_type)
-        assert isinstance(replace_entities(b'Price: &pound;100!'),  six.text_type)
-        assert isinstance(replace_entities(u'no entities'), six.text_type)
-        assert isinstance(replace_entities(u'Price: &pound;100!'),  six.text_type)
+        assert isinstance(replace_entities(b'no entities'), str)
+        assert isinstance(replace_entities(b'Price: &pound;100!'),  str)
+        assert isinstance(replace_entities(u'no entities'), str)
+        assert isinstance(replace_entities(u'Price: &pound;100!'),  str)
```
Comment (noviluni) [inline tests/test_html.py:22]: you can remove the unicode `u` from those strings

File: tests/test_html.py [hunk 3]
```diff
@@ -88,10 +95,10 @@ def test_replace_tags_multiline(self):
...
     def test_returns_unicode(self):
         # make sure it always return unicode
-        assert isinstance(remove_comments(b'without comments'), six.text_type)
-        assert isinstance(remove_comments(b'<!-- with comments -->'), six.text_type)
-        assert isinstance(remove_comments(u'without comments'), six.text_type)
-        assert isinstance(remove_comments(u'<!-- with comments -->'), six.text_type)
+        assert isinstance(remove_comments(b'without comments'), str)
+        assert isinstance(remove_comments(b'<!-- with comments -->'), str)
+        assert isinstance(remove_comments(u'without comments'), str)
+        assert isinstance(remove_comments(u'<!-- with comments -->'), str)
```
Comment (noviluni) [inline tests/test_html.py:101]: same than above (`u'`), (and the same in other files)

File: w3lib/html.py [hunk 4]
```diff
@@ -5,16 +5,16 @@
```
Comment (noviluni) [inline w3lib/html.py:5]: You can remove the encoding declaration `# -*- coding: utf-8 -*-`

**Review by noviluni — COMMENTED** [review]
Is the `tests/py3-ignores.txt` file used?

It was created in 2014. I see there was a similar file in Scrapy (https://github.com/scrapy/scrapy/commit/350aa67c3dc8997ef9d3aac9ef3f596c83758e1c), but it was used in the conftest. Where is this file used in this project? Could we remove it on this PR or do you prefer to open a new PR?

**Review by noviluni — APPROVED** [review]

**Pull request merged**